src/streamsync/ui_utils.py

```diff
@@ -0,0 +1,141 @@
+from contextvars import ContextVar
```

Comment:
I would advocate to rename this module `code_ui.py`. It contains datastructure that are useful for the rest of application. That's easy to communicate the core exist for internal purpose. We may also move into `core/ui.py` but it would mean we put the current core into `core/__init__.py.

Reply:
I'd love a better name, but I can't think of one.

I don't think `code_ui` is descriptive as it's foundational code also used by BMC.

Reply:
sorry `core_ui`

Reply:
I get it now, and I fully agree with this recommendation.